docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,360 @@
+    return values.isEmpty() ? Optional.empty() : Optional.of(values.get(values.size() - 1));
+}
+----
+
+This appears to be what we need to get a String of the remark. But what about the Index? Taking a quick peek at
+existing an `Command`...
+
+[source, java]
+.DeleteCommandParser.java
+----
+Index index = ParserUtil.parseIndex(args);
+return new DeleteCommand(index);
+----
+
+There appears to be another utility class that obtains an `Index` from the input provided by the user.
+
+Now that we have the know-how to extract the data that we need from the user's input, we can instantiate
+`RemarkCommand` by passing the data through its constructor.
+
+[source, java]
+.RemarkCommandParser.java
+----
+public RemarkCommand parse(String args) throws ParseException {
+    requireNonNull(args);
+    ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_REMARK);
+
+    Index index;
+    try {
+        index = ParserUtil.parseIndex(argMultimap.getPreamble());
+    } catch (IllegalValueException ive) {
+        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemarkCommand.MESSAGE_USAGE), ive);
+    }
+
+    String remark = argMultimap.getValue(PREFIX_REMARK).orElse("");
+
+    return new RemarkCommand(index, remark);
+}
+----
+
+NOTE: Don't forget to update `AddressBookParser` to use our new `RemarkCommandParser`!
+
+If you are stuck, check out the sample
+link:++https://github.com/j-lum/addressbook-level35/commit/c0f8419be4fb59bffafcd73026769459e5cf5c5e#diff
+-fc19ecee89c3732a62fbc8c840250508++[here].
+
+== Adding to the model
+
+Now that we have all the information that we need, let's lay the groundwork for some _persistent_ changes. We acheive
+ that by working with the `Person` model. Each field in a Person is encapsulated. This protects them from illegal
+ access and allows us to tightly couple logic specific to the field.
+
+=== Add a new `Remark` class
+
+Create a new `Remark` in `seedu.address.model.person`. Since a `Remark` is a field that is similar to `Address`, we
+can reuse a significant bit of code.
+
+A quick copy-paste and search-replace later, you should have something like
+link:++https://github.com/j-lum/addressbook-level35/commit/c8e6048f0d81f2b4b7797e838226da823892129b#diff
+-af2f075d24dfcd333876f0fbce321f25++[this].
+
+=== Make use of `Remark`
+
+Let's change `RemarkCommand` and `RemarkCommandParser` to use the new `Remark` class instead of plain `String`.
+These should be relatively simple changes.
+
+== Adding placeholder elements to the UI
+
+Without getting too deep into `fxml`, let's go on a 5 minute adventure to get some placeholder text to show up for
+each person.
+
+Simply add
+```
+@FXML
+private Label remark;
+```
+to
+link:++https://github.com/j-lum/addressbook-level35/commit/06f9897d7088d22689de478c94612f8bcbff2964#diff
+-0c6b6abcfac8c205e075294f25e851fe++[`PersonCard`]. `@FXML` is an annotation that marks a private or protected field and
```

Comment:
Try not to break a sentence over a code block.

>Simply add the following to the `foo.fxml`:
```
...
```

Reply:
Joined them.